app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt

```diff
@@ -216,6 +219,9 @@ class ChatSettingsFragment : PreferenceFragmentCompat(), ISettingsView {
 
         displayEmail.setOnPreferenceClickListener {
             settingsPresenter.loginLogout()
+            val intent = Intent(context, LoginActivity::class.java)
+            intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK
```

Comment:
Not a good idea for `FLAG_ACTIVITY_CLEAR_TOP`. What if we go back after intent, it should redirect to chat screen. Right?

Reply:
It will redirect to the settings fragment only. And I think it should redirect there only, as we have opened the login screen from there itself

Reply:
> It will redirect to the settings fragment only.

Your intent is redirecting to Login activity and clear all activities in the back stack. Right?

Reply:
Yah. Why keep the back stack filled once we are moving to the login page? Because after login the user will be redirected to the chat page. So, it's better to free up the stack

Reply:
IMO, not better user experience. Back button should redirect to chat screen instead of closing the application.

Reply:
@liveHarshit u want to say that if we press the back button at the login screen, it should redirect to the chat screen. R8?

Reply:
Yes.